aeron-driver/src/main/c/agent/aeron_driver_agent.c

```diff
@@ -546,126 +960,110 @@ int aeron_init_logging_events_interceptors(aeron_driver_context_t *context)
+    if (aeron_driver_agent_is_event_enabled(AERON_DRIVER_EVENT_SEND_CHANNEL_CLOSE))
     {
-        context->raw_log_close_func = aeron_driver_agent_raw_log_close_interceptor;
+        context->sender_proxy_on_remove_endpoint_func = aeron_driver_agent_sender_proxy_on_remove_endpoint;
     }
 
-    if (mask & AERON_RAW_LOG_FREE_OP)
+    if (aeron_driver_agent_is_event_enabled(AERON_DRIVER_EVENT_RECEIVE_CHANNEL_CREATION))
     {
-        context->raw_log_free_func = aeron_driver_agent_raw_log_free_interceptor;
+        context->receiver_proxy_on_add_endpoint_func = aeron_driver_agent_receiver_proxy_on_add_endpoint;
     }
 
-    if (mask & AERON_UNTETHERED_SUBSCRIPTION_STATE_CHANGE)
+    if (aeron_driver_agent_is_event_enabled(AERON_DRIVER_EVENT_RECEIVE_CHANNEL_CLOSE))
+    {
+        context->receiver_proxy_on_remove_endpoint_func = aeron_driver_agent_receiver_proxy_on_remove_endpoint;
+    }
+
+    if (aeron_driver_agent_is_event_enabled(AERON_DRIVER_EVENT_UNTETHERED_SUBSCRIPTION_STATE_CHANGE))
     {
         context->untethered_subscription_state_change_func =
             aeron_driver_agent_untethered_subscription_state_change_interceptor;
     }
 
+    if (aeron_driver_agent_is_event_enabled(AERON_DRIVER_EVENT_NAME_RESOLUTION_NEIGHBOR_ADDED))
+    {
+        context->name_resolution_on_neighbor_added_func = aeron_driver_agent_name_resolution_on_neighbor_added;
+    }
+
+    if (aeron_driver_agent_is_event_enabled(AERON_DRIVER_EVENT_NAME_RESOLUTION_NEIGHBOR_REMOVED))
+    {
+        context->name_resolution_on_neighbor_removed_func = aeron_driver_agent_name_resolution_on_neighbor_removed;
+    }
+
     return 0;
 }
 
 int aeron_driver_agent_context_init(aeron_driver_context_t *context)
 {
     (void)aeron_thread_once(&agent_is_initialized, initialize_agent_logging);
 
-    return aeron_init_logging_events_interceptors(context);
+    return aeron_driver_agent_init_logging_events_interceptors(context);
 }
 
-static const char *dissect_msg_type_id(int32_t id)
-{
-    switch (id)
-    {
-        case AERON_CMD_IN:
-            return "CMD_IN";
-
-        case AERON_CMD_OUT:
-            return "CMD_OUT";
-
-        case AERON_FRAME_IN:
-            return "FRAME_IN";
+#define NANOS_PER_SECOND (1000000000)
 
-        case AERON_FRAME_IN_DROPPED:
-            return "FRAME_IN_DROPPED";
-
-        case AERON_FRAME_OUT:
-            return "FRAME_OUT";
-
-        default:
-            return "unknown";
-    }
-}
-
-const char *aeron_driver_agent_dissect_timestamp(int64_t time_ms)
+const char *aeron_driver_agent_dissect_log_header(
+    const int64_t time_ns,
+    const aeron_driver_agent_event_t event_id,
+    const size_t capture_length,
+    const size_t message_length)
 {
-    static char buffer[80];
+    static char buffer[150];
 
-    snprintf(buffer, sizeof(buffer) - 1, "%" PRId64 ".%03" PRId64, time_ms / 1000, time_ms % 1000);
+    snprintf(
+        buffer,
+        sizeof(buffer) - 1,
+        "[%f] %s: %s [%zu/%zu]",
```

Comment:
`%zu` can have portability issues. Better to cast and use the PRI macros.

Reply:
Fixed.